src/panels/lovelace/editor/config-elements/hui-alarm-panel-card-editor.ts

```diff
@@ -42,177 +43,80 @@ export class HuiAlarmPanelCardEditor
-          allow-custom-entity
-        ></ha-entity-picker>
-        <paper-input
-          .label=${`${this.hass.localize(
-            "ui.panel.lovelace.editor.card.generic.name"
-          )} (${this.hass.localize(
-            "ui.panel.lovelace.editor.card.config.optional"
-          )})`}
-          .value=${this._name}
-          .configValue=${"name"}
-          @value-changed=${this._valueChanged}
-        ></paper-input>
-        <span>Used States</span> ${this._states.map(
-          (entityState, index) => html`
-            <div class="states">
-              <paper-item>${entityState}</paper-item>
-              <ha-svg-icon
-                class="deleteState"
-                .value=${index}
-                .path=${mdiClose}
-                @click=${this._stateRemoved}
-              ></ha-svg-icon>
-            </div>
-          `
-        )}
-        <mwc-select
-          .label=${this.hass.localize(
-            "ui.panel.lovelace.editor.card.alarm-panel.available_states"
-          )}
-          @selected=${this._stateAdded}
-          @closed=${stopPropagation}
-          fixedMenuPosition
-          naturalMenuWidth
-        >
-          ${states.map(
-            (entityState) =>
-              html`<mwc-list-item>${entityState}</mwc-list-item> `
-          )}
-        </mwc-select>
-        <hui-theme-select-editor
-          .hass=${this.hass}
-          .value=${this._theme}
-          .configValue=${"theme"}
-          @value-changed=${this._valueChanged}
-        ></hui-theme-select-editor>
-      </div>
+      <ha-form
+        .hass=${this.hass}
+        .data=${this._config}
+        .schema=${schema}
+        .computeLabel=${this._computeLabelCallback}
+        @value-changed=${this._valueChanged}
+      ></ha-form>
     `;
   }
 
-  static get styles(): CSSResultGroup {
-    return [
-      configElementStyle,
-      css`
-        .states {
-          display: flex;
-          flex-direction: row;
-        }
-        .deleteState {
-          visibility: hidden;
-        }
-        .states:hover > .deleteState {
-          visibility: visible;
-        }
-        ha-svg-icon {
-          padding-top: 12px;
-        }
-      `,
-    ];
-  }
+  private _valueChanged(ev: CustomEvent): void {
+    const config = ev.detail.value;
 
-  private _stateRemoved(ev: EntitiesEditorEvent): void {
-    if (!this._config || !this._states || !this.hass) {
-      return;
-    }
-
-    const target = ev.target! as EditorTarget;
-    const index = Number(target.value);
-    if (index > -1) {
-      const newStates = [...this._states];
-      newStates.splice(index, 1);
-      fireEvent(this, "config-changed", {
-        config: {
-          ...this._config,
-          states: newStates,
-        },
-      });
-    }
+    Object.keys(config).forEach(
+      (k) =>
+        (config[k] === "" ||
+          (Array.isArray(config[k]) && config[k].length === 0)) &&
+        delete config[k]
+    );
```

Comment:
This code is pretty unreadable. 
```suggestion
    for (const key of Object.keys(config)) {
      if (config[k] === "" ||
          (Array.isArray(config[k]) && config[k].length === 0))) {
        delete config[k]
      }
    };
```

Reply:
Also what fields can set empty string? in ha-form things can be undefined when not set.